homeassistant/components/rpi_i2c_ha_expanders.py

```diff
@@ -0,0 +1,208 @@
+"""
+Extends rpi_i2c_chips exapander extension for HA
+"""
+
+import logging
+import threading
+
+from .rpi_i2c_chips import MCP23018, PCF8574
+
+from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.components.binary_sensor import BinarySensorDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HAExpanderMixIn:
```

Comment:
Refactor this and avoid mixing home assistant types with device specific logic in the same class. Keep the device logic in the standalone library and just map the correct device type to corresponding home assistant entity type in home assistant.